lib/topics_filter.rb

```diff
@@ -3,8 +3,9 @@
 class TopicsFilter
   attr_reader :topic_notification_levels
 
-  def initialize(guardian:, scope: Topic.all)
-    @guardian = guardian
+  def initialize(guardian:, scope: Topic.all, loaded_topic_users_reference: false)
+    @loaded_topic_users_reference = loaded_topic_users_reference
```

Comment:
Here can't you do `@loaded_topic_users_reference = guardian.authenticated?` instead of passing it down from above? I know its tricky but this is a reeealy specific option to pass to the class